File: frontend/src/types/InspectionResult.ts
export interface InspectionResult {
  id: number;
  task_id: number;
  device_id: number;
  item_code: string;
  result_status: string; // InspectionResultStatus: NORMAL / ABNORMAL
  measured_value: string;
  photo_url: string;
  note: string;
  submitted?: boolean;
  inspector_id?: number | null;
}

export interface InspectionResultSubmitPayload {
  result_id: number;
  result_status: "NORMAL" | "ABNORMAL";
  measured_value?: string;
  note?: string;
  severity?: string;
  deadline?: string;
  owner_id?: number;
}

export interface InspectionResultSubmitResponse {
  result: InspectionResult;
  hazard: { id: number; rectify_status: string } | null;
  replayed: boolean;
}

File: backend/src/models/inspection_result.py
from pydantic import BaseModel


class InspectionResult(BaseModel):
    id: int
    task_id: int
    device_id: int
    item_code: str
    result_status: str           # NORMAL / ABNORMAL
    measured_value: str
    photo_url: str
    note: str
    created_at: str
